packages/beeq/src/components/date-picker/bq-date-picker.tsx

```diff
@@ -0,0 +1,500 @@
+ */
+@Component({
+  tag: 'bq-date-picker',
+  styleUrl: './scss/bq-date-picker.scss',
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class BqDatePicker {
+  // Own Properties
+  // ====================
+
+  private inputElem?: HTMLInputElement;
+  private labelElem?: HTMLElement;
+  private prefixElem?: HTMLElement;
+  private suffixElem?: HTMLElement;
+
+  private fallbackInputId = 'date-picker';
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqInputElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() hasLabel = false;
+  @State() hasPrefix = false;
+  @State() hasSuffix = false;
+  @State() hasValue = false;
+
+  // Public Property API
+  // ========================
+
+  /** If `true`, the Date picker input will be focused on component render */
+  @Prop({ reflect: true }) autofocus: boolean;
+
+  /** The clear button aria label */
+  @Prop({ reflect: true }) clearButtonLabel? = 'Clear value';
+
+  /**
+   * Indicates whether the Date picker input is disabled or not.
+   * If `true`, the Date picker is disabled and cannot be interacted with.
+   */
+  @Prop({ mutable: true }) disabled?: boolean = false;
+
+  /** If `true`, the clear button won't be displayed */
+  @Prop({ reflect: true }) disableClear? = false;
+
+  /** Represents the distance (gutter or margin) between the Date picker panel and the input element. */
+  @Prop({ reflect: true }) distance?: number = 8;
+
+  /** The ID of the form that the Date picker input belongs to. */
+  @Prop({ reflect: true }) form?: string;
+
+  /** The Date picker input name. */
+  @Prop({ reflect: true }) name!: string;
+
+  /** If `true`, the Date picker panel will be visible. */
+  @Prop({ reflect: true, mutable: true }) open?: boolean = false;
+
+  /** When set, it will override the height of the Date picker panel. */
+  @Prop({ reflect: true, mutable: true }) panelHeight?: string = 'auto';
+
+  /** The Date picker input placeholder text value */
+  @Prop({ reflect: true }) placeholder?: string;
+
+  /** Position of the Date picker panel */
+  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom-start';
+
+  /** Indicates whether or not the Date picker input is required to be filled out before submitting the form. */
+  @Prop({ reflect: true }) required?: boolean;
+
+  /** Represents the skidding between the Date picker panel and the input element. */
+  @Prop({ reflect: true }) skidding?: number = 0;
+
+  /** Defines the strategy to position the Date picker panel */
+  @Prop({ reflect: true }) strategy?: 'fixed' | 'absolute' = 'fixed';
+
+  /** If `true`, the Date picker panel will accepts more than 1 month to display */
+  @Prop({ reflect: true }) range: boolean = false;
+
+  /** If `true`, the Date picker panel will accepts to select multiple individual dates */
+  @Prop({ reflect: true }) multi: boolean = false;
```

Comment:
What do you think about using a single property (e.g. `type`) to set which variant of the calendar to use?
Because we cannot mix `range` with `multi` for example, so probably will be better if we just have?

```js
/** It defines how the calendar will behave, allowing single date selection, range selection, or multiple date selection */
@Prop({ reflect: true }) type: TDatePickerType = 'single';
```
where `TDatePickerType` is
```js
export const DATE_PICKER_TYPE = ['single', 'mullti', 'range'] as const;
export type TDatePickerType = (typeof DATE_PICKER_TYPE)[number];
```